apps/aecore/lib/aecore/account/account.ex

```diff
@@ -70,68 +78,56 @@ defmodule Aecore.Account.Account do
         ) :: {:ok, SignedTx.t()}
   def spend(sender, sender_priv_key, receiver, amount, fee, nonce) do
     payload = %{receiver: receiver, amount: amount}
-    spend_tx = DataTx.init(SpendTx, payload, sender, fee, nonce)
-    SignedTx.sign_tx(spend_tx, sender_priv_key)
+    spend_tx = DataTx.init(SpendTx, payload, [sender], fee, nonce)
+    SignedTx.sign_tx(spend_tx, sender, sender_priv_key)
   end
 
   @doc """
   Adds balance to a given address (public key)
   """
-  @spec transaction_in(Account.t(), integer()) :: Account.t()
-  def transaction_in(account_state, amount) do
+  @spec transaction_in!(ChainState.account(), non_neg_integer(), integer()) ::
+          ChainState.account()
+  def transaction_in!(account_state, block_height, amount) do
     new_balance = account_state.balance + amount
-    Map.put(account_state, :balance, new_balance)
-  end
-
-  @doc """
-  Deducts balance from a given address (public key)
-  """
-  @spec transaction_out(Account.t(), integer(), integer()) :: Account.t()
```

Comment:
is this removed on purpose?

Reply:
Yes, this is due to separating nonce update. I think this way is less confusing. We want to do only one nonce update per transaction always, but frequently we do many changes to account balance. (for example one fee deduction and one transfer to other account)